app/src/sharedTest/java/org/oppia/android/app/customview/LessonThumbnailImageViewTest.kt

```diff
@@ -0,0 +1,174 @@
+import org.oppia.android.app.shim.ViewBindingShimModule
+import org.oppia.android.app.testing.LessonThumbnailImageViewTestActivity
+import org.oppia.android.domain.classify.InteractionsModule
+import org.oppia.android.domain.classify.rules.continueinteraction.ContinueModule
+import org.oppia.android.domain.classify.rules.dragAndDropSortInput.DragDropSortInputModule
+import org.oppia.android.domain.classify.rules.fractioninput.FractionInputModule
+import org.oppia.android.domain.classify.rules.imageClickInput.ImageClickInputModule
+import org.oppia.android.domain.classify.rules.itemselectioninput.ItemSelectionInputModule
+import org.oppia.android.domain.classify.rules.multiplechoiceinput.MultipleChoiceInputModule
+import org.oppia.android.domain.classify.rules.numberwithunits.NumberWithUnitsRuleModule
+import org.oppia.android.domain.classify.rules.numericinput.NumericInputRuleModule
+import org.oppia.android.domain.classify.rules.ratioinput.RatioInputModule
+import org.oppia.android.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.RobolectricModule
+import org.oppia.android.testing.TestDispatcherModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.ImageLoader
+import org.oppia.android.util.parser.ImageParsingModule
+import org.oppia.android.util.parser.ImageViewTarget
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = LessonThumbnailImageViewTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class LessonThumbnailImageViewTest {
+
+  @get:Rule
+  val activityTestRule = ActivityTestRule(
+    LessonThumbnailImageViewTestActivity::class.java,
+    /* initialTouchMode= */ true,
+    /* launchActivity= */ false
+  )
+
+  @Test
+  fun callDataBindingFunctions_thenCallInject_thumbnailIsLoadedCorrectly() {
+    LessonThumbnailImageView.injectAtStart = false
```

Comment:
We shouldn't punch holes into the production class for testing purposes. Could we, instead, add the view at runtime (not through layouts), and call the bindable adapter functions before & after adding the view for each case? I think adding the view will automatically trigger injection due to the view lifecycle, and manually calling the databinding methods helps us avoid the race condition we see in production cases. Would that work?

Reply:
Done